### PR TITLE
Prevent UnicodeDecodeError in pivot_table under Py2

### LINKED ISSUE #13292: Exception when adding a unicode margins_name in pivot_table
#### Code Sample, a copy-pastable example if possible

This throws UnicodeDecodeError in python2

`pd.pivot_table(pd.DataFrame({'foo': [1,2,3]}), index=['foo'], aggfunc=len, margins=True, margins_name=u'Δοκιμή')`
#### Expected Output

Not throwing exception!
#### output of `pd.show_versions()`

<pre>
INSTALLED VERSIONS
------------------
commit: None
python: 2.7.11.final.0
python-bits: 32
OS: Windows
OS-release: 10
machine: AMD64
processor: Intel64 Family 6 Model 58 Stepping 9, GenuineIntel
byteorder: little
LC_ALL: None
LANG: None

pandas: 0.18.1
nose: None
pip: 8.1.2
setuptools: 20.10.1
Cython: None
numpy: 1.11.0
scipy: None
statsmodels: None
xarray: None
IPython: None
sphinx: None
patsy: None
dateutil: 2.5.2
pytz: 2016.4
blosc: None
bottleneck: None
tables: None
numexpr: None
matplotlib: None
openpyxl: None
xlrd: 0.9.4
xlwt: 1.0.0
xlsxwriter: None
lxml: None
bs4: None
html5lib: None
httplib2: None
apiclient: None
sqlalchemy: None
pymysql: None
psycopg2: 2.6.1 (dt dec pq3 ext)
jinja2: None
boto: None
pandas_datareader: None
</pre>

The above is a well-known problem when mixing unicode and non-unicode strings in python2: The fix (for python 2) is very simple, just add a `u` in front of the `exception_msg` string literal in https://github.com/pydata/pandas/blob/master/pandas/tools/pivot.py#L178 i.e line 178 should be `exception_msg = u'Conflicting name "{0}" in margins'.format(margins_name)`

I am not sure what's the correct way to fix that in python 3 and a python2-3 compatible way so I won't provide a PR for that.

### PR DESCRIPTION
**Pull request by mpenkov**
- [x] closes #13292 
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

**Comment by gfyoung** [comment]
@mpenkov : Will need a `whatsnew` entry

**Comment by mpenkov** [comment]
@gfyoung Added the what's new. Please check.

File: pandas/core/reshape/pivot.py [hunk 1]
```diff
@@ -145,7 +145,7 @@ def _add_margins(table, data, values, rows, cols, aggfunc,
     if not isinstance(margins_name, compat.string_types):
         raise ValueError('margins_name argument must be a string')
 
-    msg = 'Conflicting name "{name}" in margins'.format(name=margins_name)
+    msg = u('Conflicting name "{name}" in margins').format(name=margins_name)
```
Comment (TomAugspurger) [inline pandas/core/reshape/pivot.py]: The new parentheses can be removed.

File: pandas/tests/reshape/test_pivot.py [hunk 2]
```diff
@@ -1625,3 +1625,9 @@ def test_isleapyear_deprecate(self):
 
         with tm.assert_produces_warning(FutureWarning, check_stacklevel=False):
             assert isleapyear(2004)
+
+    def test_issue_13292(self):
+        # The below shouldn't raise an exception anymore.
+        frame = pd.DataFrame({'foo': [1, 2, 3]})
+        pd.pivot_table(frame, index=['foo'], aggfunc=len, margins=True,
```
Comment (jreback) [inline pandas/tests/reshape/test_pivot.py]: you don't need the comment, instead change the test name to something useful like: ``test_pivot_margins_name_unicode`` and put a comment with the issue number

File: pandas/tests/reshape/test_pivot.py [hunk 3]
```diff
@@ -1625,3 +1625,9 @@ def test_isleapyear_deprecate(self):
...
+                       margins_name=u'\u0394\u03bf\u03ba\u03b9\u03bc\u03ae')
```
Comment (jreback) [inline pandas/tests/reshape/test_pivot.py]: get the ``result = pd.pivot_table(....)``

and compare to a constructed ``expected`` with ``tm.assert_frame_equal``

File: doc/source/whatsnew/v0.20.0.txt [hunk 4]
```diff
@@ -1705,6 +1705,7 @@ Reshaping
 - Bug in ``pd.concat()`` in which concatenating with an empty dataframe with ``join='inner'`` was being improperly handled (:issue:`15328`)
 - Bug with ``sort=True`` in ``DataFrame.join`` and ``pd.merge`` when joining on indexes (:issue:`15582`)
 - Bug in ``DataFrame.nsmallest`` and ``DataFrame.nlargest`` where identical values resulted in duplicated rows (:issue:`15297`)
+- Bug in ``pivot._add_margins` when ``margins_name`` is Unicode under Py2 (:issue:`13292`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.20.0.txt]: write this from a user perspective

``Bug in :func:`pandas.pivot_table` incorrectly raising when passing unicode input for ``margins`` keyword``

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/reshape/test_pivot.py (hunk 3 above)
Comment (mpenkov) [reply pandas/tests/reshape/test_pivot.py]: @jreback Could you suggest what the constructed `expected` should look like? I'm a bit new and unsure of what should go there. Here's what I've got so far:

```python
    def test_pivot_margins_name_unicode(self):
        # issue #13292
        greek = u'\u0394\u03bf\u03ba\u03b9\u03bc\u03ae'
        frame = pd.DataFrame({'foo': [1, 2, 3]})
        table = pd.pivot_table(frame, index=['foo'], aggfunc=len, margins=True,
                               margins_name=greek)
        expected = pd.DataFrame({}, columns=['foo'], index=[1, 2, 3, greek])
        tm.assert_frame_equal(table, expected)
```

This is failing on the last line with `DataFrame shape mismatch`.

Comment (jreback) [reply pandas/tests/reshape/test_pivot.py]: don't pass the ``{}``

Comment (mpenkov) [reply pandas/tests/reshape/test_pivot.py]: Thanks for the suggestion. This is what I have now:

```python
expected = pd.DataFrame(columns=['foo'], index=[1, 2, 3, greek])
```

It's still failing:

```
E       AssertionError: DataFrame are different
E
E       DataFrame shape mismatch
E       [left]:  (4, 0)
E       [right]: (4, 1)

pandas/util/testing.py:1105: AssertionError
```

What else could be wrong?

File: pandas/tests/reshape/test_pivot.py [hunk 5]
```diff
@@ -1625,3 +1625,9 @@ def test_isleapyear_deprecate(self):
 
         with tm.assert_produces_warning(FutureWarning, check_stacklevel=False):
             assert isleapyear(2004)
+
+    def test_pivot_margins_name_unicode(self):
+        # issue #13292
+        frame = pd.DataFrame({'foo': [1, 2, 3]})
+        pd.pivot_table(frame, index=['foo'], aggfunc=len, margins=True,
```
Comment (jreback) [inline pandas/tests/reshape/test_pivot.py]: still like to compare this here

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/reshape/test_pivot.py (hunk 5 above)
Comment (mpenkov) [reply pandas/tests/reshape/test_pivot.py]: I figured out how to construct the expected object correctly. Please have a look.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @mpenkov

**Pull request merged**